doc/source/security/encrypting-temporary-files.rst

```diff
@@ -0,0 +1,71 @@
+.. _encrypting-temporary-files:
+
+==========================================================
+Encrypting Temporary Files
+==========================================================
+
+:Availability: This feature is of **Experimental** quality.
+
+For the innoDB user-created temporary tables, which are created in a temporary tablespace file, use the :variable: `innodb_temp_tablespace_encrypt` variable.
+
+.. variable:: innodb_temp_tablespace_encrypt
+
+   :cli: ``innodb-temp-tablespace-encrypt``
+   :dyn: Yes
+   :scope: Global
+   :vartype: Boolean
+   :default: ``OFF``
+
+This variable is set to ``ON``, the server encrypts the global temporary tablespace (:file: `ibtmp*` files) and the session temporary tablespaces (:file: `#innodb_temp/temp_*.ibt` files). This option does not enforce the encryption of currently open temporary files or does not rebuild the system temporary tablespace to encrypt data which has already been written.
+
+The ``CREATE TEMPORARY TABLE`` does not support the ``ENCRYPTION`` clause. The ``TABLESPACE`` clause cannot be set to innodb_temporary.
+
+The global temporary tablespace datafile ``ibtmp1`` contains temporary table undo logs while intrinsic temporary tables and user-created temporary tables are located in the encrypted session temporary tablespace.
+
+To create new temporary tablespaces unencrypted, the following variables must be set to ``OFF`` at runtime:
+
+* :variable: ``innodb_temp_tablespace_encrypt``
+
+* :variable: ``default_table_encryption``
+
+Any existing encrypted user-created temporary files and intrinsic temporary tables remain in an encrypted session.
+
+Temporary tables are only destroyed when the session is disconnected.
+
+The `default_table_encryption` setting in my.cnf determines if a temporary table is encrypted.
+
+If the `innodb_temp_tablespace` = "OFF" and the `default_table_encryption` ="ON", the user-created temporary tables are encrypted. The temporary tablespace datafile ``ibtmp1``, which contains undo logs, is not encrypted.
+
+If the ``innodb_temp_tablespace_encrypt`` is "ON" for the system tablespace, InnoDB generates an encryption key and encrypts the system temporary tablespace. If you reset the encryption to "OFF", all subsequent pages are written to an unencrypted tablespace. Any generated keys are not erased, to allow encrypted tables to be decrypted.
```

Comment:
maybe you should mention here that undo logs are encrypted

Reply:
added text about undo data.